### PR TITLE
[TASK-6945] refactor(claim): use new token selector

### PR DESCRIPTION
**Pull request by jjramirezn**
Claim flow as still using old token selector. This PR removes that token selector and uses the new one that is already used in others flows (request)

### CODE REVIEW

File: src/components/Claim/Claim.tsx [hunk 1]
```diff
@@ -133,34 +131,35 @@ export const Claim = ({}) => {
...
             if (linkDetails.claimed) {
                 setLinkState(_consts.claimLinkStateType.ALREADY_CLAIMED)
+                return
+            }
+
+            const crossChainDetails = await getCrossChainDetails(linkDetails)
+            setCrossChainDetails(crossChainDetails)
+            const tokenPrice = await utils.fetchTokenPrice(linkDetails.tokenAddress.toLowerCase(), linkDetails.chainId)
+            tokenPrice && setTokenPrice(tokenPrice?.price)
+
```
[the remark this answers is not in the document]
Comment (jjramirezn) [reply src/components/Claim/Claim.tsx:146]: they are already inside a try catch

File: src/components/Claim/Link/Initial.view.tsx [hunk 2]
```diff
@@ -134,8 +149,8 @@ export const InitialClaimLinkView = ({
                 errorMessage: '',
             })
             setLoadingState('Fetching route')
-            let tokenName = utils.getBridgeTokenName(claimLinkData.chainId, claimLinkData.tokenAddress)
-            let chainName = utils.getBridgeChainName(claimLinkData.chainId)
+            let tokenName = getBridgeTokenName(claimLinkData.chainId, claimLinkData.tokenAddress)
+            let chainName = getBridgeChainName(claimLinkData.chainId)
 
             if (tokenPrice) {
```
[the remark this answers is not in the document]
Comment (jjramirezn) [reply src/components/Claim/Link/Initial.view.tsx:155]: i like it, implementing

File: src/components/Claim/Link/Initial.view.tsx [hunk 3]
```diff
@@ -277,6 +292,19 @@ export const InitialClaimLinkView = ({
...
+            selectedChainID === claimLinkData.chainId &&
+            areTokenAddressesEqual(selectedTokenAddress, claimLinkData.tokenAddress)
+        ) {
+            setIsXChain(false)
+            setSelectedRoute(null)
+            setHasFetchedRoute(false)
+        } else {
+            setIsXChain(true)
+        }
+    }, [selectedChainID, selectedTokenAddress, claimLinkData])
```
[the remark this answers is not in the document]
Comment (jjramirezn) [reply src/components/Claim/Link/Initial.view.tsx]: i like it, implementing

File: src/components/Claim/Link/Initial.view.tsx [hunk 4]
```diff
@@ -153,8 +168,7 @@ export const InitialClaimLinkView = ({
                 }
             }
 
-            if (tokenName && chainName) {
-            } else {
+            if (!tokenName || !chainName) {
```
Comment (Hugo0) [inline src/components/Claim/Link/Initial.view.tsx:171]: lol

File: src/components/Claim/Link/Initial.view.tsx [hunk 5]
```diff
@@ -394,81 +423,15 @@ export const InitialClaimLinkView = ({
...
-                                showError: false,
-                                errorMessage: '',
-                            })
-                        }}
-                        isStatic={
-                            recipientType === 'iban' || recipientType === 'us' || !crossChainDetails ? true : false
-                        }
-                    />
+                    {recipientType !== 'iban' && recipientType !== 'us' && (
+                        <TokenSelector
```
Comment (Hugo0) [inline src/components/Claim/Link/Initial.view.tsx:427]: <3

**Review by Hugo0 — APPROVED** [review]
frontend PRs are a bit hard to review. There's so much state to think about

**Pull request merged**